### PR TITLE
Gatling generator2 issue #6927

### PR DESCRIPTION
**Pull request by ryandavis84**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR
Generates a list of gatling scenarios for every api in swagger with a type safe config.  All variables are controlled by CSV files based on the operation ID and param type.  This limits the working knowledge of gatling down to that of the business logic of a workload profile that is built out with the CSV and the configuration.
(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+        /**
+         * Model Package.  Optional, if needed, this can be used in templates
+         */
+        modelPackage = "io.swagger.client.model";
+
+        /**
+         * Reserved words.  Override this with reserved words specific to your language
+         */
+        reservedWords = new HashSet<String>(
```
Comment (jimschubert) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: I don't think this is needed.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+    /**
+     * Modifies the swagger doc to make mustache easier to use
+     *
+     * @param swagger
+     */
+    @Override
+    public void preprocessSwagger(Swagger swagger) {
+        if (swagger == null || swagger.getPaths() == null) {
+            return;
```
Comment (jimschubert) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: Does this throw elsewhere?

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+                                    }
+                                }
+                            }
+                            operation.setVendorExtension("x-gatling-body-feeder", operation.getOperationId() + "BodyFeeder");
+                            operation.setVendorExtension("x-gatling-body-feeder-params", StringUtils.join(sessionBodyVars, ","));
+                            try {
+                                FileUtils.writeStringToFile(new File(outputFolder + File.separator + dataFolder + File.separator + operation.getOperationId() + "-" + "BodyParams.csv"), StringUtils.join(bodyFeederParams, ","));
+                            } catch (IOException ioe) {
+                                LOGGER.error("Could not create feeder file for operationId" + operation.getOperationId(), ioe);
+                                System.exit(1);
```
Comment (jimschubert) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: This should throw and let the CLI or wrapping application exit.

For example, if this is consumed outside of swagger-codegen-cli, you'll be terminating the application.

Comment (ryandavis84) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: Good point i will fix this later today

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java (hunk 2 above)
Comment (ryandavis84) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: i copied this from another template once i started.  I can remove this now.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java (hunk 1 above)
Comment (ryandavis84) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: you may be right, will test with out, i think the AbstractScalaCodegen takes care of this for me now.

**Comment by ryandavis84** [comment]
Any word on this?

**Comment by jimschubert** [comment]
@ryandavis84 hey, thanks for bumping this.

Can you add a README to the generated output? I think this will make usage a little clearer for the community.

Also, when I run `gradle gatlingRun` in the generated petstore example, I get a compilation error. Seems like an escape is missing somewhere.

```

> Task :compileGatlingScala
Pruning sources from previous analysis, due to incompatible CompileSetup.
/Volumes/Extra/projects/swagger-codegen/samples/client/petstore/gatling/src/gatling/scala/io/swagger/client/model/ApiResponse.scala:14: error in interpolated string: identifier or block expected
        | "code":$code,"_type":$_type,"message":$message
                                ^
one error found


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileGatlingScala'.
> Compilation failed

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 48s
1 actionable task: 1 executed
```

Do you have the same locally?

**Comment by ryandavis84** [comment]
I will get this updated tomorrow.  Thank you.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+    }
+
+    /**
+     * Configures a friendly name for the generator.  This will be used by the generator
+     * to select the library with the -l flag.
+     *
+     * @return the friendly name for the generator
+     */
+    public String getName() {
+        return "gatling";
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: @ryandavis84 what about naming it as "scala-gatling" instead?

Naming convention for new generator: "{lang}-{framework}"

Comment (wing328) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: In other words, we'll also rename GatlingCodegen.java to ScalaGatlingCodegen.java

Comment (ryandavis84) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GatlingCodegen.java]: I will get this done this weekend.  Thank you

**Comment by wing328** [comment]
@ryandavis84 thanks for the PR, which has been merged into 2.4.0

I made some minor enhancements via 0469975632bfad1cdb7a28357f2eadff14e1956b. Please pull the latest 2.4.0 before working on other enhancements for the Gatling generator

**Pull request merged**